drivers/timer/wch_systick_ch32v00x.c

```diff
@@ -27,20 +26,36 @@
 
 #define SYSTICK ((SysTick_Type *)(DT_INST_REG_ADDR(0)))
 
+static struct k_spinlock lock;
```

Comment:
nit: the Zephyr style is verbose, and puts the local file prefix on everything including statics.

So s/lock/ch32v00x_systick_lock/

Alternatively, as there are now more than one field, put this in a `struct ch32v00x_systick_data` and have one of those.